superset-frontend/src/components/RefreshLabel/index.tsx

```diff
@@ -16,32 +16,38 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { MouseEventHandler } from 'react';
+import { useTheme } from '@superset-ui/core';
 import { Tooltip } from 'src/common/components/Tooltip';
+import Icon, { IconProps } from 'src/components/Icon';
 
-import './RefreshLabel.less';
+export interface RefreshLabelProps {
+  onClick: MouseEventHandler<SVGSVGElement>;
+  tooltipContent: string;
+}
 
-const propTypes = {
-  onClick: PropTypes.func,
-  tooltipContent: PropTypes.string.isRequired,
-};
+const RefreshLabel = ({ onClick, tooltipContent }: RefreshLabelProps) => {
+  const theme = useTheme();
 
-class RefreshLabel extends React.PureComponent {
-  render() {
-    return (
-      <Tooltip title={this.props.tooltipContent} id="cache-desc-tooltip">
-        <i
-          aria-label="Icon"
-          role="button"
-          tabIndex={0}
-          className="RefreshLabel fa fa-refresh pointer"
-          onClick={this.props.onClick}
-        />
-      </Tooltip>
-    );
-  }
-}
-RefreshLabel.propTypes = propTypes;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const IconWithoutRef = React.forwardRef((props: IconProps, ref: any) => (
+    <Icon {...props} />
+  ));
+
+  return (
+    <Tooltip title={tooltipContent}>
+      <IconWithoutRef
+        role="button"
+        onClick={onClick}
+        name="refresh"
+        css={{
```

Comment:
Wait... we're not importing `css` from `@superset-ui/core` here... is this css prop going unheard or something?

Also, if we're going to use the `css` prop, I kind of like the pattern of:
```
css={theme => css`
  cursor:  pointer;
  color: ${theme.colors.grayscale.base};
  &:hover {
    color: ${theme.colors.grayscale.base};
  }
`}
```

this just seems a little more readable to me, and uses real css (e.g. `border-color`) rather than the JS flavor (e.g. `borderColor`) and is copy/pastable to and from `styled` implementations.

Reply:
We don't need to import [`css`](https://emotion.sh/docs/css-prop) when using object styles, only when we use string styles. Babel will automatically convert this property to a `className`.

I'll disagree with you about string style vs object style. What really makes the difference for me when I chose object style is typescript checking and autocompletion. You can't do that when using string styles and that is more error-prone. Also, for me at least, is more natural coming from React and React Native style pattern.

Reply:
> About CSS-y approach I understand that it doesn't require switching the brain from CSS to JS styles but I still think that type checking is a clear winner. Let me know if you agree. Otherwise, I can revisit. Thank you for your review.

The ability to more easily port from `css` to `styled` or back was the bigger selling point, but type checking probably wins over that. Seems OK to me. 

As for autocomplete, I have an IDE plugin (`ts-styled-plugin` if memory serves) that takes care of intellisense/autocomplete with the styled/string implementation, which is helpful :)